src/adapter/graphql/src/mutations/metadata_chain_mut.rs

```diff
@@ -103,6 +135,18 @@ impl MetadataChainMut {
             Err(e @ domain::CommitError::Internal(_)) => return Err(e.int_err().into()),
         };
 
+        // TODO: encapsulate this inside dataset/chain
```

Comment:
Since `dataset.commit_event()` (unlike `chain.append()`) actually advances the reference - we know that block is the new `head` and could encapsulate the event sending logic in that function (make event bus an optional parameter).

It's still hacky, but wouldn't it be less duplication and a step closer to what we were discussing as an "end state", with dataset controlling all these things?

Reply:
Reworked. Wasn't easy, as this triggers injection of `EventBus` into `DatasetFactory` and it's methods.